src/Compilers/CSharp/Test/Semantic/Semantics/NullCheckedParameterTests.cs

```diff
@@ -791,13 +791,69 @@ class Program
 {
     static void Main(string[] args!) { }
 }";
-            var comp = CreateCompilation(source);
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
             comp.MakeMemberMissing(WellKnownMember.System_ArgumentNullException__ctorString);
             comp.MakeTypeMissing(WellKnownType.System_ArgumentNullException);
             comp.VerifyDiagnostics(
                     // (4,31): error CS0656: Missing compiler required member 'System.ArgumentNullException..ctor'
                     //     static void Main(string[] args!) { }
                     Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "args").WithArguments("System.ArgumentNullException", ".ctor").WithLocation(4, 31));
         }
+
+        [Fact]
+        public void TestNullCheckedMethodParameterWithWrongLanguageVersion()
+        {
+            var source =
+@"
+class Program
+{
+    void M(string x!) { }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp6));
+            comp.VerifyDiagnostics(
+                    // (4,20): error CS8652: The feature 'parameter null-checking' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    //     void M(string x!) { }
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, "!").WithArguments("parameter null-checking").WithLocation(4, 20));
+        }
+
+        [Fact]
+        public void TestNullCheckedLambdaParameterWithWrongLanguageVersion()
+        {
+            var source =
+@"
+using System;
+class Program
+{
+    void M()
+    {
+        Func<string, string> func = x! => x;
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp6));
+            comp.VerifyDiagnostics(
+                    // (7,38): error CS8652: The feature 'parameter null-checking' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    //         Func<string, string> func = x! => x;
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, "!").WithArguments("parameter null-checking").WithLocation(7, 38));
+        }
+
+        [Fact]
+        public void TestNullCheckedLambdaParametersWithWrongLanguageVersion()
+        {
+            var source =
+@"
+using System;
+class Program
+{
+    void M()
+    {
+        Func<string, string, string> func = (x!, y) => x;
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp6));
```

Comment:
>CSharp6 [](start = 119, length = 7)

Consider testing one of these with the highest language version not supported (that is, `CSharp8`).